aws/resource_aws_securityhub_insight.go

```diff
@@ -0,0 +1,1241 @@
+
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, securityhub.ErrCodeResourceNotFoundException) {
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error deleting Security Hub Insight (%s): %w", d.Id(), err))
+	}
+
+	return nil
+}
+
+func dateFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 20,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"date_range": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"unit": {
+								Type:         schema.TypeString,
+								Required:     true,
+								ValidateFunc: validation.StringInSlice(securityhub.DateRangeUnit_Values(), true),
+							},
+							"value": {
+								Type:     schema.TypeInt,
+								Required: true,
+							},
+						},
+					},
+				},
+				"end": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"start": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+}
+
+func ipFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 20,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"cidr": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validateCIDRNetworkAddress,
+				},
+			},
+		},
+	}
+}
+
+func keywordFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 20,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"value": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	}
+}
+
+func mapFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 20,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"comparison": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice(securityhub.MapFilterComparison_Values(), false),
+				},
+				"key": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"value": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+			},
+		},
+	}
+}
+
+func numberFilterSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 20,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"eq": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validateTypeStringNullableFloat,
+				},
+				"gte": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validateTypeStringNullableFloat,
+				},
+				"lte": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validateTypeStringNullableFloat,
+				},
+			},
+		},
+	}
+}
```

Comment:
using a string here in place of using the Float type to help distinguish between a 0 value supplied by a user and the default we get from calling d.Get()/extracting the values from the nested map..i _think_ this is an ok alternative atm, but wondering if we have other strategies to work with?